// langium/lib/grammar/generated/module.js
/******************************************************************************
 * This file was generated by langium-cli 3.3.0.
 * DO NOT EDIT MANUALLY!
 ******************************************************************************/
import { LangiumGrammarAstReflection } from '../../languages/generated/ast.js';
import { LangiumGrammarGrammar } from './grammar.js';
export const LangiumGrammarLanguageMetaData = {
    languageId: 'langium',
    fileExtensions: ['.langium'],
    caseInsensitive: false,
    mode: 'development'
};
export const LangiumGrammarParserConfig = {
    maxLookahead: 3,
};
export const LangiumGrammarGeneratedSharedModule = {
    AstReflection: () => new LangiumGrammarAstReflection()
};
export const LangiumGrammarGeneratedModule = {
    Grammar: () => LangiumGrammarGrammar(),
    LanguageMetaData: () => LangiumGrammarLanguageMetaData,
    parser: {
        ParserConfig: () => LangiumGrammarParserConfig
    }
};//# sourceMappingURL=https://main.vscode-cdn.net/sourcemaps/d2597a5c475647ed306b22de1e39853c7812d07d/node_modules/langium/lib/grammar/generated/module.js.map
