// unleash-client/lib/events.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnleashEvents = void 0;
exports.createImpressionEvent = createImpressionEvent;
// eslint-disable-next-line import/prefer-default-export
var UnleashEvents;
(function (UnleashEvents) {
    UnleashEvents["Ready"] = "ready";
    UnleashEvents["Error"] = "error";
    UnleashEvents["Warn"] = "warn";
    UnleashEvents["Unchanged"] = "unchanged";
    UnleashEvents["Changed"] = "changed";
    UnleashEvents["Synchronized"] = "synchronized";
    UnleashEvents["Count"] = "count";
    UnleashEvents["CountVariant"] = "countVariant";
    UnleashEvents["Sent"] = "sent";
    UnleashEvents["Registered"] = "registered";
    UnleashEvents["Impression"] = "impression";
    UnleashEvents["Mode"] = "mode";
})(UnleashEvents || (exports.UnleashEvents = UnleashEvents = {}));
// Wrapper to provide type checking.
function createImpressionEvent(evt) {
    return evt;
}//# sourceMappingURL=https://main.vscode-cdn.net/sourcemaps/d2597a5c475647ed306b22de1e39853c7812d07d/node_modules/unleash-client/lib/events.js.map
